impl_core/src/main/java/io/opencensus/implcore/stats/StatsComponentImplBase.java

```diff
@@ -62,18 +63,22 @@ public StatsRecorderImpl getStatsRecorder() {
 
   @Override
   public StatsCollectionState getState() {
-    return state.get();
+    return currentState.get();
   }
 
   @Override
   @SuppressWarnings("deprecation")
-  public void setState(StatsCollectionState newState) {
-    boolean stateChanged = state.set(Preconditions.checkNotNull(newState, "newState"));
-    if (stateChanged) {
-      if (newState == StatsCollectionState.DISABLED) {
-        viewManager.clearStats();
-      } else {
-        viewManager.resumeStatsCollection();
+  public synchronized void setState(StatsCollectionState newState) {
+    // Synchronized because needs to ensure that the call to resume/clear are happening atomically
+    // with the change of the state.
+    synchronized (setStateLock) {
```

Comment:
Can you please explain a bit more why block synchronization is needed here on top of method synchronization?

Reply:
Done